src/transformers/generation_flax_utils.py

```diff
@@ -238,14 +238,14 @@ def generate(
         )
         prng_key = prng_key if prng_key is not None else jax.random.PRNGKey(0)
 
-        if decoder_start_token_id is None and self.config.is_encoder_decoder:
-            raise ValueError("`decoder_start_token_id` has to be defined for encoder-decoder generation.")
-
         if self.config.is_encoder_decoder:
             # add encoder_outputs to model_kwargs
             model_kwargs = self._prepare_encoder_decoder_kwargs_for_generation(input_ids, model_kwargs)
             # prepare decoder_input_ids for generation
-            input_ids = jnp.ones((input_ids.shape[0], 1), dtype="i4") * decoder_start_token_id
+            if decoder_start_token_id is not None:
+                input_ids = jnp.ones((input_ids.shape[0], 1), dtype="i4") * decoder_start_token_id
+            else:
+                input_ids = jnp.ones((input_ids.shape[0], 1), dtype="i4")
```

Comment:
`decoder_start_token_id` should never be `None`. For mBART it is the target language id, which could be passed an argument to the `generate`. So let's remove this.